projects/igniteui-angular/src/lib/date-common/picker-base.directive.ts

```diff
@@ -29,15 +31,23 @@ export abstract class PickerBaseDirective implements IToggleView, EditorProvider
      *
      * @remarks
      * Also used as a placeholder when none is provided.
-     * Default is `"'MM/dd/yyyy'"`
      *
      * @example
      * ```html
      * <igx-date-picker inputFormat="dd/MM/yy"></igx-date-picker>
      * ```
      */
     @Input()
-    public inputFormat: string;
+    public set inputFormat(value: string) {
+        if (value) {
+            this._userSetFormat = value;
+            this._inputFormat = DateTimeUtil.getNumericInputFormat(this.locale, value);
```

Comment:
Okay, yeah I'll echo another comment from Stamen - not liking the "user set" props. Just recalculate the `_defaultInputFormat` that's already internal and keep the public API property clean and assign either one or the other to the inner input if needed.
Not super expected for the user to set "short" and read the prop and get something else. Though I'm not entirely sure the editors/pickers by design are supposed to accept that at all, in which case this functionality would not have a reason to be here. In which case this would be just a call to `updateDefaultFormat()`